infrastructures/infrastructure-azure/src/test/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructureTest.java

```diff
@@ -0,0 +1,353 @@
+                                                          "vmPassword",
+                                                          "vmPublicKey",
+                                                          "Standard_D1_v2",
+                                                          "resourceGroup",
+                                                          "region",
+                                                          "192.168.1.0/24",
+                                                          true)).thenReturn(Sets.newHashSet("123", "456"));
+
+        azureInfrastructure.acquireNode();
+
+        verify(connectorIaasController, times(1)).waitForConnectorIaasToBeUP();
+
+        verify(connectorIaasController).createAzureInfrastructure("node_source_name",
+                                                                  "clientId",
+                                                                  "secret",
+                                                                  "domain",
+                                                                  "subscriptionId",
+                                                                  "authenticationEndpoint",
+                                                                  "managementEndpoint",
+                                                                  "resourceManagerEndpoint",
+                                                                  "graphEndpoint",
+                                                                  false);
+
+        verify(connectorIaasController).createAzureInstances("node_source_name",
+                                                             "node_source_name",
+                                                             "image",
+                                                             2,
+                                                             "vmUsername",
+                                                             "vmPassword",
+                                                             "vmPublicKey",
+                                                             "Standard_D1_v2",
+                                                             "resourceGroup",
+                                                             "region",
+                                                             "192.168.1.0/24",
+                                                             true);
+
+        verify(connectorIaasController, times(2)).executeScriptWithCredentials(anyString(),
+                                                                               anyString(),
+                                                                               anyList(),
+                                                                               anyString(),
+                                                                               anyString());
+    }
+
+    @Test
+    public void testAcquireAllNodes() {
+        testAcquireNode();
```

Comment:
If you test that an exception is not thrown could you catch the exception when it's thrown and log the error however it will stop the test suite. i.e :
Exception ex;
try{
testAcquireNode();
}catch(Exception e){
 ex = e;
 log(ex)
}
assertThat(ex).isNull();